flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```
@@ -102,4 +116,18 @@ public void run(SourceContext<OUT> ctx) throws Exception {
 	public void cancel() {
 		running = false;
 	}
+
+	@Override
+	protected void acknowledgeIDs(List<Long> ids) {
+		try {
+			channel.basicAck(ids.get(ids.size() - 1), true);
+		} catch (IOException e) {
+			Log.error("Messages could not be acknowledged", e);

```

If this fails, the checkpoint is incomplete. We should throw an Exception here.